avm/res/azure-stack-hci/marketplace-gallery-image/main.bicep

```diff
@@ -0,0 +1,173 @@
+metadata name = 'Azure Stack HCI Marketplace Gallery Image'
+metadata description = 'This module deploys an Azure Stack HCI Marketplace Gallery Image.'
+
+@description('Required. Name of the resource to create.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. Resource ID of the associated custom location.')
+param customLocationResourceId string
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Required. The properties of the marketplace gallery image.')
+param marketplaceGalleryImageProperties marketplaceGalleryImagePropertiesType
```

Comment:
Good thinking,but 2nd level parameters (1st being name, tags, etc.) should be declared as parameters, allowing for defaults, documentation etc. So please just move the properties of the `marketplaceGalleryImagePropertiesType` UDT to the parameters. All other UDTs can stay as is.